src/gpio.c

```diff
@@ -0,0 +1,35 @@
+#include <stdint.h>
+
+#include "gpio.h"
+
+/**
+  * @brief  Initialize the GPIOx peripheral according to the specified parameters in the GPIO_init.
+  * @param  GPIOx where x can be (A..F) to select the GPIO peripheral for STM32F0 family.
+  * @param  GPIO_init pointer to a GPIO_init_config structure that contains
+  *         the configuration information for the specified GPIO peripheral.
+  * @retval None
+  */
+void		init_gpio(GPIO_config *GPIOx, GPIO_init_config *GPIO_init)
+{
+	GPIOx->GPIO_MODER &= ~(3U << (GPIO_init->pin * 2U));
+	GPIOx->GPIO_MODER |= (GPIO_init->mode << (GPIO_init->pin * 2U));
```

Comment:
Important: this writes the register twice. So this will always set the port to input mode first, and then set it to appropriate value.